### PR TITLE
Create CODEOWNERS

### PR DESCRIPTION
**Pull request by lbussell**

### CODE REVIEW

**Review by garath — APPROVED** [review]

**Review by MichaelSimons — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Users referenced in this file will automatically be requested as reviewers for PRs that modify the given paths.
+# See https://help.github.com/articles/about-code-owners/
+
+*                                @dotnet/dotnet-docker-reviewers
+build-info/microsoft/go-images/  @dotnet/dotnet-docker-reviewers @dagood
```
Comment (dagood) [inline CODEOWNERS:5]: https://github.com/orgs/microsoft/teams/golang-compiler for this, if cross-org is possible. (I can create a dotnet org team if needed.)

```suggestion
build-info/microsoft/go-images/  @dotnet/dotnet-docker-reviewers @microsoft/golang-compiler
```

Comment (lbussell) [reply CODEOWNERS:5]: It doesn't look like I am able to add non-dotnet GH org teams in the repo settings.

Comment (dagood) [reply CODEOWNERS:5]: Ok, created @dotnet/microsoft-go-compiler

```suggestion
build-info/microsoft/go-images/  @dotnet/dotnet-docker-reviewers @dotnet/microsoft-go-compiler
```

Comment (lbussell) [reply CODEOWNERS:5]: I gave the team write access.

**Pull request merged**